DimensionData: Adding in new methods/tests for backups

jadunham1 opened the pull request:
Added in ability to add clients to a target
Added in ability to list backup info for a target
Added in positive tests

jadunham1 committed:
DimensionData: Adding in new methods/tests for backups

tonybaloney commented on libcloud/backup/drivers/dimensiondata.py:
```diff
@@ -384,6 +421,73 @@ def cancel_target_job(self, target, job):
+        :param storage_policy: The storage policy for the client
+        :type  storage_policy: ``str``
+
+        :param schedule_policy: The storage policy for the client
```
should be "The schedule policy for the client"

tonybaloney commented on libcloud/backup/drivers/dimensiondata.py:
```diff
@@ -384,6 +421,73 @@ def cancel_target_job(self, target, job):
+        backup_elm = ET.Element('NewBackupClient',
+                                {'xmlns': BACKUP_NS})
+        ET.SubElement(backup_elm, "type").text = client
+        ET.SubElement(backup_elm, "storagePolicyName").text = storage_policy
```
Is the client code to fetch the storage policies and schedule policies implemented? It would be preferable to send the storagepolicy response from that client call instead of the name as a string.

jadunham1 replied:
I like the idea of passing both the obj and the string as done elsewhere in the DD client.  I'll make this change.

tonybaloney replied:
:+1: The list of potential storage policies alters based on the client, location and the application type :8ball:

jadunham1 committed:
Making it so we can pass in the proper classes as well as strings

jadunham1 commented:
I'm unhappy with how the classes for backups objects are directly defined here instead of in the common library.  I'm going to make this change as well.

jadunham1 committed:
DimensionData: Backups adding more tests, moving classes to common

jadunham1 commented:
Added negative tests around the info call if backups don't exist.
Moved the classes to common where the classes for the node exist.
Added a bunch of docs on the classes themselves.

jadunham1 committed:
API Doc bug, trigger is an attribute not element

jadunham1 committed:
Fixing some names/docs

tonybaloney commented:
the additional changes look great. thanks @jadunham1 merging.
